generator/views/lookml_utils.py

```diff
@@ -145,21 +147,23 @@ def _generate_dimensions_from_query(
     schema = client.query(query, job_config=job_config).schema
     dimensions = {}
     for dimension in _generate_dimensions_helper(schema or []):
-        name = dimension["name"]
+        name_key = dimension["name"]
+
+        # This prevents `time` dimension groups from overwriting other dimensions below
+        if dimension.get("type") == "time":
+            name_key += "_time"
+
         # overwrite duplicate "submission", "end", "start" dimension group, thus picking the
         # last value sorted by field name, which is submission_timestamp
         # See also https://github.com/mozilla/lookml-generator/issues/471
         if (
-            name in dimensions
-            and name != "submission"
-            and not name.endswith("end")
-            and not name.endswith("start")
-            and not (name == "event" and dimension["type"] == "time")
-            # workaround for `mozdata.firefox_desktop.desktop_installs`
-            and not (name == "attribution_dltoken" and dimension["type"] == "time")
+            name_key in dimensions
+            and dimension["name"] != "submission"
+            and not dimension["name"].endswith("end")
+            and not dimension["name"].endswith("start")
```

Comment:
Same here.

It seems like it'd be good to refactor this repeated logic in `_generate_dimensions` and `_generate_dimensions_from_query` out into a separate function named something like `_generate_dimensions_from_schema`. The only catch is the exception messages for duplicate dimensions are different, and in `_generate_dimensions` it does helpfully include the table in the error message, but for that case you could probably wrap the `_generate_dimensions_from_schema` call in a try/catch and do something like
```python
raise click.ClickException(f"Error generating dimensions for table {table!r}") from e
```